cmd/soroban-rpc/internal/db/ledger.go

```diff
@@ -103,6 +104,25 @@ func (r ledgerReader) GetLedger(ctx context.Context, sequence uint32) (xdr.Ledge
 	}
 }
 
+// BatchGetLedgers fetches ledgers in batches from the db.
+func (r ledgerReader) BatchGetLedgers(ctx context.Context, sequence uint32,
+	batchSize uint,
+) ([]xdr.LedgerCloseMeta, error) {
+	sql := sq.Select("meta").
+		From(ledgerCloseMetaTableName).
+		Where(sq.And{
+			sq.GtOrEq{"sequence": sequence},
+			sq.LtOrEq{"sequence": sequence + uint32(batchSize) - 1},
+		})
+
+	var results []xdr.LedgerCloseMeta
```

Comment:
prealloc here for sure

Reply:
Ohh yeah I already had this change done locally. That is why I had the WIP label still :p